Add bottom GradientFadeout to ListModals

itayplav opened the pull request:
### CHANGELOG

Add a gradient fadeout to the bottom of List modals

### Dependencies

<!-- Replace line with PRs which this PR depends if any --> none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [ ] Tested on Android device
- [x] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

|                                                         ContactListModal                                                         |                                                         CountryListModal                                                         | FiatListModal                                                                                                                    |
|:--------------------------------------------------------------------------------------------------------------------------------:|:--------------------------------------------------------------------------------------------------------------------------------:|----------------------------------------------------------------------------------------------------------------------------------|
| <video width="300" src="https://user-images.githubusercontent.com/4023066/219825478-3012cf0e-5988-4746-a5b1-ee82f8bd6b77.mp4" /> | <video width="300" src="https://user-images.githubusercontent.com/4023066/219825484-7dcf1b6d-dd78-4d4e-9a39-20b5d0813626.mp4" /> | <video width="300" src="https://user-images.githubusercontent.com/4023066/219825488-65d7e33e-0c32-4fee-a6da-fce4d57ca39a.mp4" /> |


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203476138018936

paullinator commented on src/components/modals/GradientFadeout.tsx:
```diff
@@ -0,0 +1,34 @@
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const color = theme.modal
+  const colors: string[] = MARKS.map(mark => color + `0${Math.floor(255 * mark).toString(16)}`.slice(-2))
```
I realized I missed this on the login-ui PR, but colors should be memoized otherwise you're constantly re-rendering.

paullinator commented on src/components/modals/ListModal.tsx:
```diff
@@ -94,7 +95,8 @@ export function ListModal<T>({
       />
-      {closeArrow && <ModalCloseArrow onPress={handleCancel} />}
+      <GradientFadeOut />
+      {!closeArrow ? null : <ModalCloseArrow onPress={handleCancel} />}
```
Fadeout should just be put into the close arrow

itayplav replied:
Adding Fadout to all `ModalCloseArrow` components will introduce a bug where the `GradientFadeout` covers overlay buttons in all `ButtonsModal` components, since those use `ModalCloseArrow`, in additional to one-off modals such as the `ScanModal`. This visual feature (in it's current state) is just meant for list modals.

paullinator replied:
In that case, put the fadeout underneath the close arrow.

paullinator commented on src/__tests__/modals/__snapshots__/AccelerateTxModal.test.tsx.snap:
```diff
@@ -53,7 +53,7 @@ exports[`AccelerateTxModalComponent should render with loading props 1`] = `
       showSpinner={false}
     />
   </View>
-  <ModalCloseArrow
```
Do not make this a fixup commit. Renames should be a commit of their own.

itayplav replied:
Now a seperate commit

paullinator commented:
/rebase

itayplav commented:
Updated to incorporate conflicts with https://github.com/EdgeApp/edge-react-gui/pull/4039

Notable changes from approved PR:

- Use `marginBottom: theme.rem(-1)` as part of [two nested views](https://github.com/EdgeApp/edge-react-gui/pull/4014/commits/005e9b62b2cc008708ff62eba0eee5920a2d5e1a#diff-d80bcec443955f2f976bf509b2ac2c0f4194350a1ca379fb0904ba3d573fef63R49-R50) in the `ModalFooter` to anchor the fadeout effect to the bottom, instead of the previous implementations method of supplying `paddingRem=[1, 1, 0, 1]` to `ListModal` and `ThemedModal`
- Added `accessibilityHint="Close Modal"` to the `<TouchableOpacity>`
